modules/katamari/src/main/ts/ephox/katamari/api/Id.ts

```diff
@@ -22,3 +24,21 @@ export const generate = (prefix: string): string => {
 
   return prefix + '_' + random + unique + String(time);
 };
+
+/**
+ * Generate a uuidv4 string
+ * In accordance with RFC 4122 (https://datatracker.ietf.org/doc/html/rfc4122)
+ */
+const uuidV4 = (): `${string}-${string}-${string}-${string}-${string}` => {
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  if (Type.isFunction(window.crypto.randomUUID) && window.isSecureContext) {
```

Comment:
I know you like to be defensive, but the second check implies the function exists on all supported browsers. On older browsers that don't have `crypto`, `window.isSecureContext` is likely to be `undefined`.
```suggestion
  if (window.isSecureContext) {
```